share/functions/__fish_anyeditor.fish

```diff
@@ -0,0 +1,16 @@
+function __fish_anyeditor --description "Print a editor to use, or an error message"
+    set -l editor
+    if set -q VISUAL
+        echo $VISUAL | read -at editor
+    else if set -q EDITOR
+        echo $EDITOR | read -at editor
+    else
+        echo
+        echo (_ 'External editor requested but $VISUAL or $EDITOR not set.')
+        echo (_ 'Please set VISUAL or EDITOR to your preferred editor.')
```

Comment:
Any particular reason this doesn't just go to stderr?

Reply:
yeah, stderr is better here